src/Rewriter.hs

```diff
@@ -28,13 +25,26 @@ import Text.Printf
 import Yaml (ExtraArgument (..))
 import qualified Yaml as Y
 
+-- | Function type for saving intermediate rewriting steps.
+type SaveStepFunc = Program -> Integer -> IO ()
```

Comment:
@yarkivaev let's move this `SaveStepFunc` and `createSaveStepFunc` to a new separated module, let's call it `Steps`. Here you'll have  `type SaveStepFunc`, `saveStep` and `dontSaveStep` functions which are exported to all other modules which need it and there are no more duplications of `noSaveStep` in `Dataize` module and tests